### PR TITLE
Cad example work

### PR DESCRIPTION
**Pull request by Erfan-Ahmadi**

### CODE REVIEW

File: common/CommonAPI.h [hunk 1]
```diff
@@ -1,20 +1,29 @@
 #ifndef __NBL_COMMON_API_H_INCLUDED__
 #define __NBL_COMMON_API_H_INCLUDED__
 
+#define _NBL_STATIC_LIB_
```
Comment (devshgraphicsprogramming) [inline common/CommonAPI.h:4]: can you live without this?

Comment (Erfan-Ahmadi) [reply common/CommonAPI.h:4]: don't remember what that is, but when I remove it, it still works :D

Comment (devshgraphicsprogramming) [reply common/CommonAPI.h:4]: remove it, its a define that should only be defined when CMake is in the static config, shoul dbe already in `BuildConfig.h` which is autogeneretated

Comment (AnastaZIuk) [reply common/CommonAPI.h:4]: I believe a lot of examples were defining it, may be a remnant of very old code. `_NBL_SHARED_BUILD_` is defined instead https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/config/BuildConfigOptions.h.in#L90

**Pull request merged**